htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -256,6 +256,36 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 				return -2;
 			}
 
+
+			// check unicity for input serial numbered equipments ( different for lots managed products)
+			if ( $product->status_batch == 2 && ($type == 0 || $type == 3) )
```

Comment:
Why moving this test at beginning.
Testing at end of code (when every action were done, like triggers for example) is more secured.

Reply:
it seemed to me it is a gain of performance to  first check the passed parameter before doing other checks into database.

Reply:
@eldy what do you think of my argument for performance ? is it ok for you or do i have to put the code below ?

Reply:
Performance is important but it will lead to a lot of other problem. But integrity too.
And checking integrity of data at begin of transaction means all the rest of transaction can break integrity.
And integrity has priority versus performance. So test at end (where it is) must remain.